pkg/framework/aries/framework.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package aries
+
+import (
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/context"
+	errors "golang.org/x/xerrors"
+)
+
+// Aries provides access to clients being managed by the framework.
+type Aries struct {
+	transport api.TransportProviderFactory
+}
+
+// Option configures the framework.
+type Option func(opts *Aries) error
+
+// New initializes the Aries framework based on the set of options provided.
+func New(opts ...Option) (*Aries, error) {
+	// get the default framework options
+	defOpts := defFrameworkOpts()
```

Comment:
Just a thought.
Not sure about having default factory settings as opts, if someone passes custom opts then we may have duplicate opts. It may work here in this case due to order of opts. Need proper test cases to see if overriding actually works (I can see in one of the test you passed mock transport factory), just to be at safe side we shouldn't pass default settings as opts. 

Instead at line number 28, initialize Aries struct with default opts.

Reply:
This was done based on [comments](https://github.com/hyperledger/aries-framework-go/pull/88#discussion_r312992096).

Reply:
I will mark it as resolved

Reply:
Yes order of opts needs to be kept intact.